### PR TITLE
tiny js world 

### PR DESCRIPTION
**Pull request by SergeyShytikov**
### tiny JS world 

[Demo](https://sergeyshytikov.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/SergeyShytikov/a-tiny-JS-world/blob/master/index.js)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/SergeyShytikov/tiny-js-world-task/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+
+function getPropertiesValueToString(obj) {
+return print(Object.values(obj).join('; '))
+}
```
Comment (OleksiyRudenko) [inline submissions/SergeyShytikov/tiny-js-world-task/index.js:66]: Looks like a generic method to convert any object into a string.

However, as the task requires specific order of properties listed `Object` methods like `keys`, `values`, `entries` shouldn't be used as these do not guarantee any particular order of keys/values.

Once you solve this the method will stop being generic but rather specific to our domain (Tiny JS World and its inhabitants). Thus you will need to reflect this specificity in function and its parameter names.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@SergeyShytikov good job!
Please check the comment below.

File: submissions/SergeyShytikov/tiny-js-world-task/index.js (hunk 1 above)
Comment (SergeyShytikov) [reply submissions/SergeyShytikov/tiny-js-world-task/index.js]: Thanks for the advice, for some reason I did not read all the requirements of the task.

File: submissions/SergeyShytikov/tiny-js-world-task/index.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+
+const mappedInhabitants = inhabitants.map(obj => keysOrder.map(key => obj[key])) ;
+mappedInhabitants.forEach(item => print(item.join('; ')));
```
Comment (OleksiyRudenko) [inline submissions/SergeyShytikov/tiny-js-world-task/index.js:74]: If we chain `inhabitants.map` and following `forEach` we do not need an extra variable.
`obj` is too generic. As a peer developer I want to **read** instantly what objects we work here with.
Same is about `item`.

Do not use generic variable names unless it is a function specifically designed to work with multiple types of instances (e.g. we are building a generic data processing library).

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@SergeyShytikov well done!
A few things to improve and we are done.

File: submissions/SergeyShytikov/tiny-js-world-task/index.js (hunk 2 above)
Comment (SergeyShytikov) [reply submissions/SergeyShytikov/tiny-js-world-task/index.js]: You right, after a while I won't be able to figure out what's going on here, either.

**Review by OleksiyRudenko — APPROVED** [review]
@SergeyShytikov well done, grats!

P.S. Please install prettier.io plugin on your IDE for it to take care of formatting and code style for you. On bigger codebase it is worth it definitely.

**Pull request merged**